src/cargo/util/command_prelude.rs

```diff
@@ -558,12 +558,19 @@ Run `{cmd}` to see possible targets."
 
     fn get_profile_name(
         &self,
-        gctx: &GlobalContext,
         default: &str,
         profile_checking: ProfileChecking,
     ) -> CargoResult<InternedString> {
         let specified_profile = self._value_of("profile");
 
+        let err_message = |flag: &str| {
+            format!(
+                "\
+the `--{flag}` flag can not be specified with the `--profile` flag
+Please remove one of the flags."
+            )
+        };
```

Comment:
This should work and below we can just call `check_conflict(flag)?` instead.

```suggestion
        fn check_conflict(flag: &str) -> CargoResult<()> {
            bail!(
                "\
the `--{flag}` flag can not be specified with the `--profile` flag
Please remove one of the flags."
            )
        };
```